beacon-chain/rpc/validator/proposer.go

```diff
@@ -284,19 +284,9 @@ func (vs *Server) deposits(ctx context.Context, currentVote *ethpb.Eth1Data) ([]
 		return []*ethpb.Deposit{}, nil
 	}
 
-	upToEth1DataDeposits := vs.DepositFetcher.AllDeposits(ctx, canonicalEth1DataHeight)
-	depositData := [][]byte{}
-	for _, dep := range upToEth1DataDeposits {
-		depHash, err := ssz.HashTreeRoot(dep.Data)
-		if err != nil {
-			return nil, errors.Wrap(err, "could not hash deposit data")
-		}
-		depositData = append(depositData, depHash[:])
-	}
-
-	depositTrie, err := trieutil.GenerateTrieFromItems(depositData, int(params.BeaconConfig().DepositContractTreeDepth))
```

Comment:
can you gate this via a feature flag ? Since this affects block production, it would be better to be safe here.